src/Datadog.Trace.ClrProfiler.Native/cor_profiler.cpp

```diff
@@ -854,17 +874,10 @@ HRESULT CorProfiler::GenerateVoidILStartupMethod(const ModuleID module_id,
   const auto assembly_emit =
       metadata_interfaces.As<IMetaDataAssemblyEmit>(IID_IMetaDataAssemblyEmit);
 
-  // TODO fix this for .NET Core
-  // Define an AssemblyRef to mscorlib, needed to create TypeRefs later
   mdModuleRef mscorlib_ref;
-  ASSEMBLYMETADATA metadata{};
-  metadata.usMajorVersion = 4;
-  metadata.usMinorVersion = 0;
-  metadata.usBuildNumber = 0;
-  metadata.usRevisionNumber = 0;
-  BYTE public_key[] = {0xB7, 0x7A, 0x5C, 0x56, 0x19, 0x34, 0xE0, 0x89};
-  assembly_emit->DefineAssemblyRef(public_key, sizeof(public_key), "mscorlib"_W.c_str(),
-                                   &metadata, NULL, 0, 0, &mscorlib_ref);
+  if (!CreateAssemblyRefToMscorlib(assembly_emit, &mscorlib_ref)) {
+    Warn("GenerateVoidILStartupMethod: CreateAssemblyRefToMscorlib failed.");
```

Comment:
What's the fallout if this fails?

Reply:
We've never seen this fail because we should always be able to reference mscorlib and the base types. It's possible this would fail badly though because the IL method we dynamically generate would be incorrect. We could return S_FAIL here on the bad case?

Reply:
I added a code path in ce97f2383821373e1045dfbe5f035fe88c000ddf to exit immediately if it fails.